### PR TITLE
fix: fix safe_issubclass for NewType

### PR DESCRIPTION
**Pull request by Czaki**
Closes https://github.com/napari/napari/issues/7129

Today I learn that such code:

```python 

new_int = NewType("new_int", int)

issubclass(new_int, new_int)
```

Raises exception instead of returning True. 

So this PR is a followup to #665  to fully resolve the upstream napari bug.

### CODE REVIEW

File: tests/test_util.py [hunk 1]
```diff
@@ -80,3 +80,12 @@ def test_subclass_future(self):
...
         assert not safe_issubclass(Future[list[int]], Future[list[str]])
+
+    def test_subclass_new_type(self):
+        new_int = typing.NewType("new_int", int)
+
+        assert safe_issubclass(new_int, new_int)
+        assert safe_issubclass(list[new_int], list[new_int])
+        assert safe_issubclass(list[new_int], list[new_int])
+        assert safe_issubclass(list[new_int], typing.Sequence[new_int])
+        assert safe_issubclass(list[new_int], list[new_int])
```
Comment (jni) [inline tests/test_util.py:91]: Why is this assert repeated three times @Czaki? And should one also test typing.List?

Comment (Czaki) [reply tests/test_util.py:91]: ruff....

**Comment by Czaki** [comment]
@tlambert03 The #671 changed type annotation used in `tests/test_util.py` making many test repetitive. 
Should we restore these changes or simplify tests?

File: tests/test_util.py (hunk 1 above)
Comment (jni) [reply tests/test_util.py:91]: 🤣 that is amazing

File: src/magicgui/_util.py [hunk 2]
```diff
@@ -187,6 +187,8 @@ def _safe_isinstance_tuple(obj: object, superclass: object) -> bool:
 
 def safe_issubclass(obj: object, superclass: object) -> bool:
     """Safely check if obj is a subclass of superclass."""
+    if obj == superclass:
```
Comment (tlambert03) [inline src/magicgui/_util.py:190]: when comparing types, shouldn't we use `is` ?

**Comment by tlambert03** [comment]
> Should we restore these changes or simplify tests?

ah, i think these should probably be restored.  since `typing.List` is still not an alias for `list` (in fact they don't even equal each other), it probably doesn't hurt to just keep checking that the typing objects work explicitly

File: src/magicgui/_util.py (hunk 2 above)
Comment (Czaki) [reply src/magicgui/_util.py:190]:
```suggestion
    if obj is superclass:
```
lets check

**Comment by tlambert03** [comment]
Thanks!

**Comment by tlambert03** [comment]
Do you need this released asap?  Can it wait a week or so to give time for testing the current pre-release?  Or would you prefer a cherry pick back-release

**Comment by Czaki** [comment]
This is a question to @psobolewskiPhD

**Comment by psobolewskiPhD** [comment]
I mean it's been around and it only affects things that didn't update to drop older python, so I don't think it's critical?
I think being able to update the image.sc thread that it's been fixed and say look for a new magicgui release in a week or 2 should be fine?

**Comment by tlambert03** [comment]
Sounds good thanks

**Pull request merged**